frontend/pages/hosts/details/cards/Scripts/Scripts.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useContext, useRef, useState } from "react";
+import { useQuery } from "react-query";
+import { AxiosResponse } from "axios";
+import { InjectedRouter } from "react-router";
+
+import PATHS from "router/paths";
+import scriptsAPI, {
+  IHostScript,
+  IHostScriptsResponse,
+} from "services/entities/scripts";
+import { IApiError, IError } from "interfaces/errors";
+import { NotificationContext } from "context/notification";
+
+import Card from "components/Card";
+import TableContainer from "components/TableContainer";
+import EmptyTable from "components/EmptyTable";
+import DataError from "components/DataError";
+import { ITableQueryData } from "components/TableContainer/TableContainer";
+import ScriptDetailsModal from "pages/DashboardPage/cards/ActivityFeed/components/ScriptDetailsModal";
+
+import { generateDataSet, generateTableHeaders } from "./ScriptsTableConfig";
+
+const baseClass = "host-scripts-section";
+
+interface IScriptsProps {
+  hostId?: number;
+  page?: number;
+  isHostOnline: boolean;
+  router: InjectedRouter;
+}
+
+const Scripts = ({ hostId, page = 0, isHostOnline, router }: IScriptsProps) => {
+  const [showScriptDetailsModal, setShowScriptDetailsModal] = useState(false);
+  // used to track the current script execution id we want to show in the show
+  // details modal.
+  const scriptExecutionId = useRef<string | null>(null);
+
+  const { renderFlash } = useContext(NotificationContext);
+
+  const {
+    data: hostScriptResponse,
+    isLoading: isLoadingScriptData,
+    isError: isErrorScriptData,
+    refetch: refetchScriptsData,
+  } = useQuery<IHostScriptsResponse, IError>(
+    ["scripts", hostId, page],
+    () => scriptsAPI.getHostScripts(hostId as number, page),
+    {
+      refetchOnWindowFocus: false,
+      retry: false,
+      enabled: Boolean(hostId),
+    }
+  );
+
+  if (!hostId) return null;
+
+  const onQueryChange = (data: ITableQueryData) => {
+    router.push(`${PATHS.HOST_SCRIPTS(hostId)}?page=${data.pageIndex}`);
+  };
+
+  const onActionSelection = async (action: string, script: IHostScript) => {
+    switch (action) {
+      case "showDetails":
+        if (!script.last_execution) return;
+        scriptExecutionId.current = script.last_execution.execution_id;
+        setShowScriptDetailsModal(true);
+        break;
+      case "run":
+        try {
+          await scriptsAPI.runScript(script.script_id);
+          refetchScriptsData();
+        } catch (e) {
+          const error = e as AxiosResponse<IApiError>;
+          renderFlash("error", error.data.errors[0].reason);
```

Comment:
Should we refetch the scripts data here also? In that case, maybe move the refetch into a finally block?